### PR TITLE
Fix new Cluster Regex

### PR DESCRIPTION
**Pull request by jyash97**

### CODE REVIEW

File: src/pages/ClusterPage/new.js [hunk 1]
```diff
@@ -210,7 +210,7 @@ export default class NewCluster extends Component {
 		let pattern = /^[a-zA-Z0-9]+([-]+[a-zA-Z0-9]*)*[a-zA-Z0-9]+$/;
 		if (provider === 'gke') {
 			// gke cluster names can't start with a number, no capital letters allowed
-			pattern = /^[a-z]+([-]+[a-z0-9]*)*[a-z0-9]+$/;
+			pattern = /^[a-z]+([-a-z0-9]*)*[a-z0-9]+$/;
```
Comment (siddharthlatest) [inline src/pages/ClusterPage/new.js:213]: Why do we have the last `[a-z0-9]+` condition? That requires two or more characters. And doesn't allow last character to be a `-`. I don't think this is required.

File: src/pages/ClusterPage/new.js [hunk 2]
```diff
@@ -210,7 +210,7 @@ export default class NewCluster extends Component {
 		let pattern = /^[a-zA-Z0-9]+([-]+[a-zA-Z0-9]*)*[a-zA-Z0-9]+$/;
```
Comment (siddharthlatest) [inline src/pages/ClusterPage/new.js:210]: The general pattern as well can avoid using the last `[a-zA-Z0-9]+` condition, I think so.

**Review by siddharthlatest — CHANGES_REQUESTED** [review]
There are some unintended side-effects of the current regex pattern.

**Comment by siddharthlatest** [comment]
@raazcrzy Can you confirm on this?

File: src/pages/ClusterPage/new.js (hunk 1 above)
Comment (octate) [reply src/pages/ClusterPage/new.js:213]: Naming convention for GKE:
- Name must be lowercase letters, numbers, and hyphens
- Name must start with a lowercase letter
- Name must end with a lowercase letter or a number.

**Comment by siddharthlatest** [comment]
@jyash97 We can merge this PR in that case. It seems last char can't be a hyphen.

**Review by siddharthlatest — APPROVED** [review]

**Pull request merged**